pkg/binary/envoy/debug/pii.go

```diff
@@ -0,0 +1,72 @@
+package debug
+
+import (
+	"fmt"
+	"bitbucket.org/creachadair/shell"
+	"crypto/sha256"
+)
+
+// filter the array of logs using formatStr and validFieldNames, hash the resulting array of logs and return the final array
+func process(logsArr []string, formatStr string, validFieldNames map[string]bool) (processedLogs []string) {
+	defer func() {
+		if r := recover(); r != nil {
+			fmt.Println("process func panicking, recovering")
+		}
+	} ()
+
+	fieldNames, ok := shell.Split(formatStr)
```

Comment:
The reason of why I put `fieldNames` rather than `fields` is that I'm trying to differentiate between names of the field and values of the field, i.e. field name of the field value `[2019-09-05T17:53:36.908Z]` is `[%START_TIME%]`

Reply:
Am I understanding it wrongly?